### PR TITLE
Various UI Changes + Skin Randomizer

### PR DESCRIPTION
**Pull request by TsFreddie**
Sorry this is unprompted. Just a few small changes plus reimpletmented #2613 (because I forgot to create new branches)

Changes:

- Add missing Localizations for "Please enter the password." and skin parts catagories (due to missing context)
- Allow listbox to have no header and remove the listbox's header in skin part selection which is a duplicated info.
- Make scrollregion's scrollbar more integrated into the content.
  - basically removed this corner gap image
- Change demo list's column header to CORNER_ALL to make it the same as server browser's header.
- Added skinparts' name into the selection list as well, because why not, we have a search bar on top.
- Implemented a cleaner version of the skin randomizer.

closes #2613

### CODE REVIEW

**Comment by TsFreddie** [comment]
I also prepopulated the editbox after clicking save in the custom skin menu to have the skin name. So you can edit a skin and quickly save it without having to type the skin name again.

If you click save with a standard skin, it will populated the editbox with `copy_{skinname}` instead.
